content/kb/troubleshooting-restore-pvc-stuck-using-velero-csi-plugin-version-lower-than-0.4.md

```diff
@@ -0,0 +1,32 @@
+---
+title: "Troubleshooting: Restoring PersistentVolumeClaim is stuck in the Pending state via Velero CSI Plugin < 0.4"
+author: Ray Chang
+draft: false
+date: 2022-12-15
+categories:
+  - "restore"
+---
+
+## Applicable versions
+
+All Longhorn versions.
+
+## Symptoms
+
+When using Velero with Velero CSI Plugin (< 0.4) to restore PersistentVolumeClaim manipulated by Longhorn, the PVC will be stuck in the `Pending` state.
+
+## Reason
+
+For Longhorn versions using `longhornio/csi-provisioner:v2.1.2`, when it processes a PVC to provision the volume, Longhorn CSI provisioner will only recognize the `volume.beta.kubernetes.io/storage-provisioner` annotation which will be tagged to each PVC via Kubernetes after determining the storage provisioner. This annotation of the PVC will be backed up together via Velero.
+
+After restoring the PVC via Velero with its CSI plugin (< 0.4), it will only remove the `volume.beta.kubernetes.io/storage-provisioner` but keep the `volume.kubernetes.io/storage-provisioner` annotation intact from the PVC, because the plugin doesn't respect the general available `volume.kubernetes.io/storage-provisioner` annotation. Thus, it will cause the restoring PVC will be failed to be processed by the built-in Longhorn CSI provisioner and be stuck in the `Pending` state.
```

Comment:
```suggestion
After restoring the PVC via Velero with its CSI plugin (< 0.4), it will only remove the `volume.beta.kubernetes.io/storage-provisioner` but keep the `volume.kubernetes.io/storage-provisioner` annotation intact from the PVC, because the plugin doesn't respect the general available `volume.kubernetes.io/storage-provisioner` annotation. Because Kubernetes will not add `volume.kubernetes.io/storage-provisioner` to the PVC which already has the beta annotation, it will cause the restoring PVC will be failed to be processed by the built-in Longhorn CSI provisioner and be stuck in the `Pending` state.
```